### ISSUE TITLE
Number of variables in LPA

### ISSUE THREAD

**Issue by DavidLRodrigues** [hyunsooseol/snowRMM#26]
Hi, I'm running a LPA on jamovi with the snowRMM module. In total, I have 19 variables, but I can only get results when I use 15 variables. Once I enter the 16th variable, the program doesn't give me any result from the analysis. Is there a maximum number os valuables to be considered when using this module?

**Comment by hyunsooseol** [comment]
Hi [DavidLRodrigues](https://github.com/DavidLRodrigues)

Thanks for your question. I am using **tidyLPA** R package for LPA analysis. I don't think maximum number of variables to be considered for analysis. If possible, could you share .omv file in here or mail at **snow@cau.ac.kr** to see if what happens ?

Kind Regards
Seol

**Comment by DavidLRodrigues** [comment]
Hu @hyunsooseol 
thank you for your quick reply. I sent the .omv file to the email you provided.
Thank you in advance for the help!

Best regards,
David

**Comment by hyunsooseol** [comment]
Hi [DavidLRodrigues](https://github.com/DavidLRodrigues)

The issue doesn't seem to be the number of variables. 
1. Computational Limitations: LPA involves complex calculations, especially when the number of variables increases. The computational load increases exponentially with more variables, which can lead to memory issues or extended processing times that might cause the analysis to fail or not return results. This can be particularly challenging if your system has limited computational resources (e.g., RAM or processing power).
2. Model Complexity: With more variables, the model becomes more complex, and it may become difficult for the algorithm to converge to a solution. The estimation process in LPA involves finding the best-fitting model, which can be problematic with a high number of variables due to increased parameter estimation complexity.

In conclusion, I recommend taking this data and running it as an R package. If you see the same symptoms in R, it is most likely the problem mentioned above.

Kind Regards
Seol

**Comment by hyunsooseol** [comment]
Hi [DavidLRodrigues](https://github.com/DavidLRodrigues)

Here is R codes with your data. but it was not working.
![image](https://github.com/user-attachments/assets/428e04c4-4dc4-4309-aa51-258ae35670e9)

Kind Regards
Seol

**Comment by hyunsooseol** [comment]
Hi
My guess is that there is an issue with your model settings. I don't think there is a problem with jamovi at this stage. Have you try to run with R codes that I written above?

**Comment by DavidLRodrigues** [comment]
I did and I get the same results as you, which is very odd. I will need to dig deeper on this to try to figure out what is going on. Thank you so much for your help!

**Comment by hyunsooseol** [comment]
Hi
Have you try to run with example data called 'iris'? Overall Model Fit option provide '**best fit**' ,`not estimate_profiles`, provided by Model Fit options(Model1, 2, 3,6). 
Here is a result of iris data below.
![image](https://github.com/user-attachments/assets/851e941e-99c0-450f-b53f-831b9de8c881)
Therfore, I think, It seems to take a long time to find the best fit with many variables because it doesn't converge.

**Comment by DavidLRodrigues** [comment]
I ran with another dataset of mine and the results were similar. Can you please why does jamovi compute Model Fit for six models (Model 1, 2, 3, and 6) each with a given number of latent classes?

**Comment by hyunsooseol** [comment]
When you look at instructions in results screen, **mclust** R package only provide 4 models. Further, The LPA module first computes the **best fit** and then outputs specific **Model Fit** and **Estimates** values. However, since the **best fit** is not calculated with your data, no other analysis results are provided afterward.

**Comment by hyunsooseol** [comment]
If I didn't add an overall model fit to the LPA analysis, users would have to find the best class and model one by one. This is very inconvenient for users, so I added the Overall Model fit option.

**Comment by hyunsooseol** [comment]
This is a Best model fit R codes with iris data--
# Best Model fit with 4 models(1,2,3,6)---------
```
res<- iris %>% 
   select(Sepal.Length, Sepal.Width, Petal.Length, Petal.Width) %>% 
   estimate_profiles(n_profiles = 2:4, models = c(1, 2, 3, 6)) %>% 
   get_fit() %>% 
   as.data.frame()
 res
```
